### DISCUSSION TITLE
Help Offered

### DISCUSSION THREAD

**Discussion by opengoalapp** [DonsetPG/narya#34] [General]
Hello all,

Just wanted to post to offer any assistance with this project that you think would be of best use to move things forward, including with the corner tracking idea posted by @larsmaurath.

I'm probably intermediate level Python (e.g. can get pretty much anything to work with the help of stack overflow, but not really production level code) and Keras (but zero pytorch experience). I'm familiar with, but not particularly experienced with the core CV ideas/algorithms and libraries like opencv. I've got access to a mid-range laptop GPU but otherwise would use Colab as their GPU seems about 20% faster than mine for most model training, so I don't know whether that setup is suitable?

A couple of random things:

- Are you aware of the [SoccerNet](https://soccer-net.org/) data set? There's 550 matches available with various types of labelling. There is a very simple NDA to sign (I've signed it) and the only stipulation for working in groups is that everyone signs it.
- I've been playing around with some of my own match footage. The homography estimation is working great around the penalty areas (lots of keypoints I'd imagine) but less so around the centre circle. This I'm guessing is partly down to a mismatch in the pitch dimensions and the template i.e. the penalty area dimensions are the same for any pitch, but the distance between edge of the area to halfway line/touchline is variable. Have you thought about ways of handling that or do you just need an appropriate template for each video clip?

Anyway, congrats on the project so far. It is really impressive what you have managed to achieve, and as mentioned would be more than happy to help out in any way I can.

Charlie

**Comment by DonsetPG** [comment]
Hey,
Thanks a lot for your message I appreciate it! 

> Just wanted to post to offer any assistance with this project that you think would be of best use to move things forward, including with the corner tracking idea posted by @larsmaurath.

I am starting to run out of time so if you want to help with the player labelling, feel free to help! 

> Are you aware of the SoccerNet data set? There's 550 matches available with various types of labelling. There is a very simple NDA to sign (I've signed it) and the only stipulation for working in groups is that everyone signs it.

Yes but I did not dive that much into it. In your opinion, what could be cool to do with it?

>  but less so around the centre circle.

I think its mainly because if the fact that we have much less key points in the center of the field (something that could be solved with a uniform grid).

**Reply by opengoalapp** [reply]
> I am starting to run out of time so if you want to help with the player labelling, feel free to help!

Great, I will take a look at what you have already. I've played around with the labelling tools mentioned in this repo so hopefully I should be good to go.

> Yes but I did not dive that much into it. In your opinion, what could be cool to do with it?

I don't have any revolutionary ideas, it was just more about maybe diversifying the training sets (as I saw you mentioned something about getting videos from YouTube/Footballia) and with that collection you could generate a batch of frames from a large number of games/camera setups if finding and processing match footage was time consuming. Also using their labels maybe the Narya models could be extended in the future to detect and process the core match events such as passing, shooting etc.

**Reply by larsmaurath** [reply]
@alen1010 @opengoalapp 

I added an overview table of which tasks are claimed and which ones are still open. Also added some non-labeling items that may be interesting. 

https://github.com/DonsetPG/narya/discussions/27#discussioncomment-632838

**Reply by larsmaurath** [reply]
@opengoalapp SoccerNet sounds like a great resource for match footage, thanks for sharing! Do I understand it correctly that they have labels for Actions and Cameras (but not player tracking or homography)?

**Reply by opengoalapp** [reply]
@alen1010 @larsmaurath 

Table and other ideas look great. Thanks

There does also appear to be some homography matrices for each frame of the high quality videos in the calibration.json files of each video folder. I haven't looked at them properly or tried to do any warping with them. I also couldn't find any mention of them in the docs. So not sure how useful it is.
